x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -154,7 +154,8 @@ protected static List<Batch<LogicalPlan>> rules() {
             new SubstituteSurrogates(),
             new ReplaceRegexMatch(),
             new ReplaceNestedExpressionWithEval(),
-            new ReplaceAliasingEvalWithProject()
+            new ReplaceAliasingEvalWithProject(),
+            new SkipQueryOnEmptyMappings()
```

Comment:
Because this can be determine upfront, which facilitates the rest of the rules and folding of LocalRelation through-out the plan.
Skip is done at the end for cases where a normal relationship is changed due to filtering.

Reply:
It does make the other rules do less work, yes, but since there's a `ONCE` limiter, it might then have made more sense as a first rule. Anyways, minor.